Mathlib/Topology/Algebra/InfiniteSum/Constructions.lean

```diff
@@ -73,6 +73,27 @@ section ContinuousMul
 
 variable [CommMonoid α] [TopologicalSpace α] [ContinuousMul α]
 
+section Sum
+
+@[to_additive]
+lemma HasProd.sum {α β M : Type*} [CommMonoid M] [TopologicalSpace M] [ContinuousMul M]
+    {f : α ⊕ β → M} {a b : M}
+    (h₁ : HasProd (f ∘ Sum.inl) a) (h₂ : HasProd (f ∘ Sum.inr) b) : HasProd f (a * b) := by
+  have : Tendsto ((∏ b ∈ ·, f b) ∘ sumEquiv.symm) (atTop.map sumEquiv) (nhds (a * b)) := by
+    rw [Finset.sumEquiv.map_atTop, ← prod_atTop_atTop_eq]
+    convert (tendsto_mul.comp (nhds_prod_eq (x := a) (y := b) ▸ Tendsto.prod_map h₁ h₂))
+    ext s
+    simp
+  simpa [Tendsto, ← Filter.map_map] using this
+
+@[to_additive]
+lemma Multipliable.sum {α β M : Type*} [CommMonoid M] [TopologicalSpace M] [ContinuousMul M]
+    (f : α ⊕ β → M) (h₁ : Multipliable (f ∘ Sum.inl)) (h₂ : Multipliable (f ∘ Sum.inr)) :
+    Multipliable f :=
+  ⟨_, .sum h₁.choose_spec h₂.choose_spec⟩
```

Comment:
IMO it's nicer to do this by breaking h1 and h2 into HasProd rather than choice, but I won't insist on it.

Reply:
Is this what you were thinking of?

Reply:
Yup!